docs/slotfilling.rst

```diff
@@ -192,3 +192,254 @@ like ``MyCustomFormField`` and overriding the ``validate()`` method.
+                                     "ask_utt": "utter_ask_cuisine",
+                                     "clarify_utt": "utter_explain_cuisine_restaurant"
+                                     },
+                         "people": {
+                                    "ask_utt": "utter_ask_people",
+                                    "clarify_utt": "utter_explain_people_restaurant"
+                                    },
+                         "location": {
+                                      "ask_utt": "utter_ask_location",
+                                      "clarify_utt": "utter_explain_location_restaurant"
+                                      }
+                         }
+
+            finish_action = "deactivate_form"
+
+            exit_dict = {
+                         "goodbye": "deactivate_form",
+                         "request_hotel": "deactivate_form_switch"
+                         }
+
+            chitchat_dict = {"chitchat": "utter_chitchat"}
+
+            details_intent = "utter_ask_details"
+
+            rules = {
+                     "cuisine":{
+                                "mcdonalds": {
+                                              'need':['location'],
+                                              'lose':['people', 'price']
+                                             }
+                                }
+                    }
+
+            failure_action = 'utter_human_hand_off'
+            super(RestaurantForm, self).__init__(name, slot_dict, finish_action, exit_dict, chitchat_dict, details_intent, rules, failure_action=failure_action)
+The extra arguments not defined above are defined below, but are optional.
+Stories
+~~~~~~~
+
+We also need to let Rasa Core predict when to activate the Forms. We do this by defining an action (in this bot example ``activate_restaurant`` and ``activate_hotel``) which contains a ``StartForm`` event. We then write stories where this action is triggered:
+
+.. code-block:: md
+
+    ## Generated Story 6817547858592778997
+    * request_restaurant
+        - activate_restaurant
+        - slot{"switch": false}
+        - slot{"cuisine": "mexican"}
+        - slot{"form_complete": false}
+        - utter_happy
+    * chitchat
+        - utter_chitchat
+    * request_hotel
+        - activate_hotel
+
+
+which allows core to predict when to activate the restaurant. Using a combination of the ``finish_action`` or ``exit_dict`` one can featurize the way that the form finished. In this case, we set a slot to say after an exit whether the form had been completed or not (``form_complete``). It is important to note that *how* the slots were filled within the form is not featurized. Simply which slots are filled at the time of the form's deactivation is relevant and helps downstream core predictions. We see above an example of a story where the form has not been filled and the user has exited. We also have a story where the slots of the form have all been filled:
```

Comment:
language here can be simplified as well, try to avoid phrases like "It is important to note that"